lot_service/src/main/kotlin/ru/zveron/service/LotService.kt

```diff
@@ -0,0 +1,174 @@
+                price = price,
+                lotFormId = lotFormId,
+                dateCreation = Instant.now(),
+                status = LotStatus.ACTIVE,
+                gender = gender.toGender(),
+                sellerId = seller.id,
+                categoryId = categoryId,
+                channelType = communicationChannelList.toChannelType(),
+                addressId = addressId
+            )
+        }
+
+        with(lot) {
+            // Инициализируем по умолчанию статистику с нулевыми параметрами
+            statistics = LotStatistics(lot = lot)
+            photos = request.photosList
+                .map { LotPhoto(lot = lot, imageId = it.id, order = it.order) }
+            parameters = request.parametersMap.map {
+                LotParameter(
+                    id = LotParameter.LotParameterKey(it.key, lot.id),
+                    value = it.value,
+                    lot = lot
+                )
+            }
+        }
+
+        return lotRepository.save(lot)
+    }
+
+
+    @Transactional
+    fun editLot(lot: Lot, request: EditLotRequest, seller: SellerProfile): Lot {
+        validateRequestForUpdating(request.photosList, request.communicationChannelList, seller)
+
+        with(lot) {
+            title = request.title
+            description = request.description
+            price = request.price
+            channelType = request.communicationChannelList.toChannelType()
+            parameters = request.parametersMap.map {
+                LotParameter(
+                    id = LotParameter.LotParameterKey(it.key, lot.id),
+                    value = it.value,
+                    lot = lot
+                )
+            }
+            photos = request.photosList.map {
+                LotPhoto(
+                    lot = lot,
+                    imageId = it.id,
+                    order = it.order
+                )
+            }
+        }
+
+        return lotRepository.saveAndFlush(lot)
+    }
+
+    fun closeLot(lot: Lot, request: CloseLotRequest) {
+        if (lot.status != LotStatus.ACTIVE) {
+            throw LotException(Status.INTERNAL, "you cant close inactive lot")
+        }
+
+        when (request.closingLotReason) {
+            ClosingLotReason.SOLD -> {
+                if (!request.hasCustomerId()) {
+                    throw LotException(Status.INVALID_ARGUMENT, "cant closing lot without field customer id")
+                }
+
+                if (request.customerId == lot.sellerId) {
+                    throw LotException(Status.INVALID_ARGUMENT, "cant choose myself as customer, because it your lot")
+                }
+                // TODO add logic for possible customers -> add reviews ZV-300
+                lot.status = LotStatus.CLOSED
+            }
+
+            else -> {
+                lot.status = LotStatus.CANCELED
+            }
+        }
+
+
+        lotRepository.save(lot)
+    }
+
+    private fun validateRequestForUpdating(
```

Comment:
Раз это используется не только в обновлении, но и в создании
```suggestion
    private fun validateLotProperties(
```